### PR TITLE
Feature/hooks translation

### LINKED ISSUE #5: Translate Section 2: Hooks, useState, useReducer, useEffect, useRef and useImperativeHandle
Please feel free to ask to work on this issue and I'll assign it to you 😄

https://github.com/typescript-cheatsheets/react#hooks

### PR DESCRIPTION
**Pull request by bnrosa**
Added the Hooks section.
- Some of the words that were keywords or technical terms were kept in English or the English term is added after the translation between parenthesis.
- All code was kept the same, comments were translated.

Resolves #5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -331,3 +331,210 @@ const MyArrayComponent = () => (Array(5).fill(<div />) as any) as JSX.Element;
...
+```
+
+Veja também no artigo em inglês (utilizando [Using Inferred Types](https://react-typescript-cheatsheet.netlify.app/docs/basic/troubleshooting/types/#using-inferred-types) se precisar usar um tipo complexo para o qual você depende da inferência.
+
+No entanto, muitos hooks são inicializados com valores nulos e você pode se perguntar como deve fazer para definir o tipo. Declare explicitamente o tipo e use um tipo de união (union type):
+
+```tsx
+const [user, setUser] = React.useState<IUser | null>(null);
+
+// mais tarde...
```
Comment (GiselaMD) [inline README.md]: Maybe we could have this as "mais adiante..."? What do you think?

File: README.md [hunk 2]
```diff
@@ -331,3 +331,210 @@ const MyArrayComponent = () => (Array(5).fill(<div />) as any) as JSX.Element;
...
+Você também pode usar asserções de tipo (type assertions) se um estado for inicializado logo após o setup e sempre tiver um valor definido após o setup:
+
+```tsx
+const [user, setUser] = React.useState<IUser>({} as IUser);
+
+// mais tarde...
+setUser(newUser);
+```
+
+"Mentimos" temporariamente para o compilador de Typesctipr que `{}` é do tipo `IUser`. Você deve então configurar o estado de `user` — se não o fizer, o resto do seu código pode depender do fato de que `user` é do tipo `IUser` e isso pode levar a erros em tempo de execução (runtime errors).
```
Comment (GiselaMD) [inline README.md]: Can you fix typo here? `Typesctipr`

**Review by GiselaMD — COMMENTED** [review]
Thanks for the translation! Just added a few comments/suggestions.

File: README.md (hunk 1 above)
Comment (bnrosa) [reply README.md]: Sure, it's a better term.

File: README.md (hunk 2 above)
Comment (bnrosa) [reply README.md]: Of course.

**Comment by swyxio** [comment]
very awesome!

**Review by GiselaMD — APPROVED** [review]

**Pull request merged**